test/integration/targets/lookup_first_found/tasks/main.yml

```diff
@@ -38,6 +38,30 @@
       - result.results | length == 1
       - result.results[0].item.endswith('/foo1')
 
+- name: roles search for files in the files subdir
+  debug:
+    msg: "{{ lookup('first_found', 'findme.txt') }}"
```

Comment:
instead of using file name as verification, use unique token per location (same file name) and inspect that the content of the file is the correct one.